prime-angular-frontend/src/app/modules/site-registration/pages/organization-claim-page/organization-claim-page.component.ts

```diff
@@ -0,0 +1,94 @@
+import { ToastService } from '@core/services/toast.service';
+
+import { Subscription } from 'rxjs';
+import { exhaustMap, map } from 'rxjs/operators';
+
+import { RouteUtils } from '@lib/utils/route-utils.class';
+
+import { AuthService } from '@auth/shared/services/auth.service';
+import { SiteRoutes } from '@registration/site-registration.routes';
+import { OrganizationFormStateService } from '@registration/shared/services/organization-form-state.service';
+import { OrganizationClaimFormModel } from '@registration/shared/models/organization-claim-form.model';
+import { OrganizationClaimPageFormState } from './organization-claim-page-form-state.class';
+import { BcscUser } from '@auth/shared/models/bcsc-user.model';
+import { Organization } from '@registration/shared/models/organization.model';
+import { OrganizationAgreementViewModel } from '@shared/models/agreement.model';
+
+@Component({
+  selector: 'app-organization-claim-page',
+  templateUrl: './organization-claim-page.component.html',
+  styleUrls: ['./organization-claim-page.component.scss']
+})
+export class OrganizationClaimPageComponent implements OnInit {
+  public formState: OrganizationClaimPageFormState;
+  public title: string;
+  public routeUtils: RouteUtils;
+  public isCompleted: boolean;
+  public isClaimExistingOrg: boolean;
+  public busy: Subscription;
+
+  constructor(
+    private formUtilsService: FormUtilsService,
+    private organizationFormStateService: OrganizationFormStateService,
+    private organizationResource: OrganizationResource,
+    private toastService: ToastService,
+    private authService: AuthService,
+    private route: ActivatedRoute,
+    router: Router
+  ) {
+    this.title = this.route.snapshot.data.title;
+    this.routeUtils = new RouteUtils(route, router, SiteRoutes.MODULE_PATH);
+    this.isClaimExistingOrg = false;
+  }
+
+  public onBack(): void {
+    this.routeUtils.routeWithin(SiteRoutes.COLLECTION_NOTICE);
+  }
+
+  public onContinue(): void {
+    // Claim an existing organization
+    if (this.isClaimExistingOrg) {
+      if (this.formUtilsService.checkValidity(this.formState.form)) {
+        // check if the organization w/ the PEC has already been claimed
+        this.busy = this.organizationResource.getOrganizationClaim({ pec: this.formState.json.pec })
+          .subscribe((result: boolean) => {
+            if (result) {
+              this.toastService.openErrorToast(`The organization associated the site of PEC code ${this.formState.json.pec} cannot be claimed.`);
+            }
+            else {
+              this.routeUtils.routeRelativeTo([SiteRoutes.SITE_MANAGEMENT], { queryParams: { claimOrg: true } });
+            }
+          });
+      }
+      else {
+        // invalid form
+        return;
+      }
+    }
+    else {
+      this.routeUtils.routeRelativeTo([SiteRoutes.SITE_MANAGEMENT], { queryParams: { claimOrg: false } });
```

Comment:
Kind of still need the claimOrg=false explicitly since the guard is using it to determine the routes for first time visiter.  Thoughts?

In manageNoOrganizationRouting of OrganizationGuard
1. claimOrg=undefined, go to organization claim page
2. claimOrg query param exists (true or false), go to signing authority page

Reply:
I would move the claim page to the other side of organization signing authority then the flow would be:

1. Access signing authority and create the signing authority, and then route to claim using initial route flow - `/organizations/0/organization-signing-authority`
2. Allow the now existing signing authority to claim an existing organization, otherwise route to organization name - `/organizations/0/claim`.  Using this route pattern will allow us to reuse the claim page in the future and not just on initial workflow.
3. Allow the signing authority to fill out the form, and on submit create the organization instead of creating it in the organization signing authority page - `/organizations/0/organization-name`.  Which once the organization is created would replace the oid with the proper oid from creation and carry on with registration.

We'll be allowing signing authority updates in the future, and this would keep each page with a single responsibility.  Also, we are claiming an organization so the pages should be within the organization routes.